src/logic/contracts.ts

```diff
@@ -0,0 +1,85 @@
+import { ChainId, ChainName, getChainName } from '@dcl/schemas'
+import { AppComponents } from '../types'
+import { ContractsResponse } from '../types/contracts'
+
+const getCollectionsQuery = `
+  query getCollections {
+    collections(first: 1000) {
+      id
+    }
+  }
+`
+
+export async function isValidContractAddress(
+  components: Pick<AppComponents, 'config' | 'fetcher' | 'collectionsSubgraph'>,
+  address: string
+): Promise<boolean> {
+  return (
+    (await isWhitelisted(components, address)) ||
+    (await isCollectionAddress(components, address))
+  )
+}
+
+async function isWhitelisted(
+  components: Pick<AppComponents, 'config' | 'fetcher'>,
+  address: string
+): Promise<boolean> {
+  const {
+    config,
+    fetcher: { fetch },
+  } = components
+
+  const contractAddressesURL = await config.requireString(
+    'CONTRACT_ADDRESSES_URL'
+  )
+  const chainName = await getCollectionChainName(components)
+
+  const remoteResult = await fetch(contractAddressesURL, {
```

Comment:
Can we save this in mem in order to not download it at every request? It would be great if we can update it at every hour or something in background